components/restapi/core/component.go

```diff
@@ -40,7 +40,7 @@ type dependencies struct {
 	AppInfo          *app.Info
 	RestRouteManager *restapipkg.RestRouteManager
 	Protocol         *protocolpkg.Protocol
```

Comment:
Do we still need the Protocol dep?

Reply:
We can do without it actually by just adding one more method for IsNodeSynced, so I have done that.